virttest/env_process.py

```diff
@@ -98,10 +99,39 @@ def preprocess_vm(test, params, env, name):
     :param env: The environment (a dict-like object).
     :param name: The name of the VM object.
     """
+    def validate_memory_resource():
+        """Validate host has enough memory to lauch VMs"""
+        vms = set(params.objects("vms")) - set(env.get_all_vms())
+        magnification, requried_mem, count = 1.0, 0, 1
+        for vm_name in vms:
+            vm_params = params.object_params(vm_name)
+            if vm_params.get("start_vm") == "yes":
+                mem = "%sM" % vm_params.get("mem", 512)
+                requried_mem += float(utils_misc.normalize_data_size(
+                    mem, order_magnitude="K"))
+                count += 1
+        if (params.get("setup_ksm") == "yes" and
+                params.get("ksm_run", "1") == "1"):
+            magnification = 1.2
+        provide_mem = "%s KB" % (memory.freememtotal() * magnification)
+        provide_mem = float(
+            utils_misc.normalize_data_size(provide_mem))
+        # make memory size aligned to 256Mib
+        suggest_mem = int(provide_mem / count / 256) * 256 + 256
+
+        return (requried_mem > provide_mem, suggest_mem)
```

Comment:
When compare required_mem to provide_mem, the unit of required_mem is KB, but provide_mem is MB which has been convert in line 117, so required_mem is always > provide_mem